core/src/main/scala/cats/data/EitherT.scala

```diff
@@ -174,6 +245,194 @@ final case class EitherT[F[_], A, B](value: F[Either[A, B]]) {
+   * @see [[redeem]] and [[redeemWith]] for optimizations on `attempt.map`
+   *      and `attempt.flatMap` respectively
+   */
+  def attempt(implicit F: Functor[F]): EitherT[F, A, Either[A, B]] =
+    EitherT.right(value)
+
+  /**
+   * Handles errors thrown in `F[_]` (if it implements [[ApplicativeError]]),
+   * by materializing them into `Either` values.
+   *
+   * $monadErrorF
+   *
+   * Example:
+   * {{{
+   *   import cats.effect.IO
+   *
+   *   val rio = EitherT.pure[IO, String](10)
+   *
+   *   // Yields Right(10)
+   *   rio.attemptF
+   *
+   *   val dummy = new RuntimeException("dummy")
+   *   val lio = EitherT[IO, String, Int](IO.raiseError(dummy))
+   *
+   *   // Yields Left(RuntimeException("dummy"))
+   *   lio.attemptF
+   * }}}
+   *
+   * Note that in this sample we are materializing the `Throwable` of the
+   * underlying `IO`, even the source `EitherT` value is not parametrized
+   * with it.
+   *
+   * @see [[attempt]] for recovering errors expressed via `EitherT`
+   *
+   * @see [[redeemF]] and [[redeemWithF]] for optimizations on `attemptF.map`
+   *      and `attemptF.flatMap` respectively
+   */
+  def attemptF[E](implicit F: ApplicativeError[F, E]): EitherT[F, A, Either[E, B]] =
+    EitherT(F.map(F.attempt(value)) {
+      case Right(Right(b)) => Right(Right(b))
+      case Right(Left(a))  => Left(a)
+      case Left(e)         => Right(Left(e))
+    })
+
+  /**
+   * Handle any error, potentially recovering from it, by mapping it via
+   * the provided function.
+   *
+   * This is the implementation of [[ApplicativeError.handleError]].
+   *
+   * Example:
+   * {{{
+   *   parseNum(s).handleError(_ => 0L)
+   * }}}
+   *
+   * @see [[handleErrorWith]] for recovering errors by mapping them to
+   *      `EitherT` values (aka the equivalent of `flatMap` for errors)
+   *
+   * @see [[handleErrorF]] for recovering errors thrown in `F[_]`
+   */
+  def handleError(f: A => B)(implicit F: Functor[F]): EitherT[F, A, B] =
+    EitherT(F.map(value) {
+      case Left(a) => Right(f(a))
+      // N.B. pattern match does not do `case Right(_)` on purpose
+      case right => right
+    })
+
+  /**
+   * Handles any error in `F[_]`, potentially recovering from it, by mapping
+   * it via the provided function.
+   *
+   * $monadErrorF
+   *
+   * Example:
+   * {{{
+   *   import cats.effect.IO
+   *   import java.lang.Long.{parseNum => javaParseNum}
+   *
+   *   def parseNum(s: String, r: Int = 10): IO[Long] =
+   *     IO(javaParseNum(s, r))
+   *
+   *   val eio = EitherT.right[String](parseNum("invalid"))
+   *
+   *   // Yields 0L on evaluation
+   *   eio.handleErrorF(_ => 0L)
+   * }}}
+   *
+   * @see [[handleErrorWithF]] for recovering errors by mapping them to
+   *      `EitherT` values (aka the equivalent of `flatMap` for errors)
+   *
+   * @see [[handleError]] for recovering errors expressed via `EitherT`
+   */
+  def handleErrorF[E](f: E => B)(implicit F: ApplicativeError[F, E]): EitherT[F, A, B] =
```

Comment:
Same comment about A

Reply:
I guess we can do `A1 >: A` or something.